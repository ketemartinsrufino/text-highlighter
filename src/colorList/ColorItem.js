import { useState } from 'react';
import styled from 'styled-components';

const ColorItemStyled = styled.div`
    border: ${({isSelected}) => isSelected ? 2: 1}px black solid;
    height:30px;
    width: 50px;
    background-color: ${({color}) => color};

    :hover {
        cursor: pointer:
    }
`;

const ColorItem = ({color}) => {

    return <ColorItemStyled 
            color="red" 
            isSelected={this.isSelected("red")}
            onClick={this.getOnClick("red")}
        />
}