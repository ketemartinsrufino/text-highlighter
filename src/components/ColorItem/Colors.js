export const COLORS = {
  red: '#ef5350', 
  yellow: '#ffeb3b', 
  green: '#8eda8e'
}