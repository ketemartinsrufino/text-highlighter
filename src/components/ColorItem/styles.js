import styled from 'styled-components';

export const ColorItemStyled = styled.div`
  background-color: ${({color}) => color};
  border-radius: 50%;
  height: 30px;
  width: 30px;
  margin: 0 5px;
  border: 1px #a97c7c solid;

  ${({ isSelected }) => isSelected && `
    border: 2px black solid
  `}

  :hover {
    cursor: pointer:
  }
`;