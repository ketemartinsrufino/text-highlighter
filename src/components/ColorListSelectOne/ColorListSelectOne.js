import React from 'react';
import ColorList from '../ColorList/ColorList';
import PropTypes from 'prop-types';

const ColorListSelectOne = ({selectedColor, onColorSelect}) => (
  <ColorList
    selectedColors={[selectedColor]}
    onClick={onColorSelect}
  />
);

ColorListSelectOne.defaults = {
  selectedColor: ''
}

ColorListSelectOne.propTypes = {
  selectedColor: PropTypes.string,
  onColorSelect: PropTypes.func.isRequired
};

export default ColorListSelectOne;