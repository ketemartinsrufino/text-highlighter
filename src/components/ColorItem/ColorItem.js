import React from 'react';
import PropTypes from 'prop-types';
import { ColorItemStyled } from './styles';

const ColorItem = ({ color, isSelected, onClick }) => (
  <ColorItemStyled
    color={color}
    isSelected={isSelected}
    onClick={() => onClick(color)}
    className={`ColorItem${isSelected ? ' selected' : ''}`}
  />
)

ColorItem.defaultProps = {
  isSelected: false
}

ColorItem.propTypes = {
  isSelected: PropTypes.bool,
  color: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}

export default ColorItem;