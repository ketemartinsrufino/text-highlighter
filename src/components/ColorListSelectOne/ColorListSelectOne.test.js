import React from 'react';
import { shallow, mount, render } from 'enzyme';
import ColorListSelectOne from './ColorListSelectOne';
import toJson from 'enzyme-to-json';
import { COLORS } from '../ColorItem/Colors';
import ColorItem from '../ColorItem/ColorItem';

describe('ColorListSelectOne', () => {
  it('is defined with a default value', () => {
    const component = shallow(
      <ColorListSelectOne onColorSelect={jest.fn()}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('receive a "selectedColor"', () => {
    const component = shallow(
      <ColorListSelectOne 
        selectedColor={COLORS.yellow} 
        onColorSelect={jest.fn()}/>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('receive a "selectedColor" and select another color', () => {
    const component = mount(
      <ColorListSelectOne 
        selectedColor={COLORS.yellow} 
        onColorSelect={jest.fn()}/>
    );

    component.find(ColorItem).first().simulate('click');
    expect(toJson(component)).toMatchSnapshot();
  });
  
  it('only one color can be selected at the same time', () => {
    const component = mount(
      <ColorListSelectOne 
        selectedColor={COLORS.yellow} 
        onColorSelect={jest.fn()}/>
    );
    
    component.find(`div.ColorItem[color="${COLORS.red}"]`).first().simulate('click');
    component.find(`div.ColorItem[color="${COLORS.yellow}"]`).first().simulate('click');

    const selected = component.find('div.ColorItem.selected');
    expect(selected.length).toEqual(1);
    expect(selected.prop('color')).toEqual(COLORS.yellow);
  });

});