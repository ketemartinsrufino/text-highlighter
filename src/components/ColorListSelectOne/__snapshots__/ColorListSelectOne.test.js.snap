// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorListSelectOne is defined with a default value 1`] = `
<ColorList
  colors={
    Array [
      "#ef5350",
      "#ffeb3b",
      "#8eda8e",
    ]
  }
  onClick={[MockFunction]}
  selectedColors={
    Array [
      undefined,
    ]
  }
/>
`;

exports[`ColorListSelectOne receive a "selectedColor" 1`] = `
<ColorList
  colors={
    Array [
      "#ef5350",
      "#ffeb3b",
      "#8eda8e",
    ]
  }
  onClick={[MockFunction]}
  selectedColors={
    Array [
      "#ffeb3b",
    ]
  }
/>
`;

exports[`ColorListSelectOne receive a "selectedColor" and select another color 1`] = `
<ColorListSelectOne
  onColorSelect={
    [MockFunction] {
      "calls": Array [
        Array [
          "#ef5350",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  selectedColor="#ffeb3b"
>
  <ColorList
    colors={
      Array [
        "#ef5350",
        "#ffeb3b",
        "#8eda8e",
      ]
    }
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [
            "#ef5350",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    selectedColors={
      Array [
        "#ffeb3b",
      ]
    }
  >
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "lastClassName": "fGHxMu",
              "rules": Array [
                "
  display: flex;
  margin: 10px 0;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="sc-bdVaJa fGHxMu"
        >
          <ColorItem
            color="#ef5350"
            isSelected={false}
            key="color-#ef5350"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [
                    "#ef5350",
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
          >
            <styled.div
              className="ColorItem"
              color="#ef5350"
              isSelected={false}
              onClick={[Function]}
            >
              <StyledComponent
                className="ColorItem"
                color="#ef5350"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "FVvLF",
                      "rules": Array [
                        "
  background-color: ",
                        [Function],
                        ";
  border-radius: 50%;
  height: 30px;
  width: 30px;
  margin: 0 5px;
  border: 1px #a97c7c solid;

  ",
                        [Function],
                        "

  :hover {
    cursor: pointer;
  }
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                isSelected={false}
                onClick={[Function]}
              >
                <div
                  className="ColorItem sc-bwzfXH kStvIh"
                  color="#ef5350"
                  onClick={[Function]}
                />
              </StyledComponent>
            </styled.div>
          </ColorItem>
          <ColorItem
            color="#ffeb3b"
            isSelected={true}
            key="color-#ffeb3b"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [
                    "#ef5350",
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
          >
            <styled.div
              className="ColorItem selected"
              color="#ffeb3b"
              isSelected={true}
              onClick={[Function]}
            >
              <StyledComponent
                className="ColorItem selected"
                color="#ffeb3b"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "FVvLF",
                      "rules": Array [
                        "
  background-color: ",
                        [Function],
                        ";
  border-radius: 50%;
  height: 30px;
  width: 30px;
  margin: 0 5px;
  border: 1px #a97c7c solid;

  ",
                        [Function],
                        "

  :hover {
    cursor: pointer;
  }
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                isSelected={true}
                onClick={[Function]}
              >
                <div
                  className="ColorItem selected sc-bwzfXH FeDsH"
                  color="#ffeb3b"
                  onClick={[Function]}
                />
              </StyledComponent>
            </styled.div>
          </ColorItem>
          <ColorItem
            color="#8eda8e"
            isSelected={false}
            key="color-#8eda8e"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [
                    "#ef5350",
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
          >
            <styled.div
              className="ColorItem"
              color="#8eda8e"
              isSelected={false}
              onClick={[Function]}
            >
              <StyledComponent
                className="ColorItem"
                color="#8eda8e"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "FVvLF",
                      "rules": Array [
                        "
  background-color: ",
                        [Function],
                        ";
  border-radius: 50%;
  height: 30px;
  width: 30px;
  margin: 0 5px;
  border: 1px #a97c7c solid;

  ",
                        [Function],
                        "

  :hover {
    cursor: pointer;
  }
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                isSelected={false}
                onClick={[Function]}
              >
                <div
                  className="ColorItem sc-bwzfXH FVvLF"
                  color="#8eda8e"
                  onClick={[Function]}
                />
              </StyledComponent>
            </styled.div>
          </ColorItem>
        </div>
      </StyledComponent>
    </styled.div>
  </ColorList>
</ColorListSelectOne>
`;
